src/shims/unix/fs.rs

```diff
@@ -1441,8 +1419,7 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         if let IsolatedOp::Reject(reject_with) = this.machine.isolated_op {
             this.reject_in_isolation("`readlink`", reject_with)?;
             let eacc = this.eval_libc("EACCES");
-            this.set_last_error(eacc)?;
-            return Ok(-1);
+            return Ok(this.set_last_error_and_return_i32(eacc));
```

Comment:
It seems like previously it returns ``Ok(-1)`` but now it returns ``Ok(Ok(-1))``?